lib/logmsg/nvhandle-descriptors.h

```diff
@@ -34,4 +34,18 @@ struct _NVHandleDesc
   guint8 name_len;
 };
 
+typedef struct
+{
+  NVHandleDesc *data;
+  guint   len;
+  guint   alloc;
+} NVHandleDescArray;
+
+#define NVHANDLE_DESC_ARRAY_INITIAL_SIZE 256
+
+NVHandleDescArray *nvhandle_desc_array_new();
+void nvhandle_desc_array_free(NVHandleDescArray *self);
+void nvhandle_desc_array_append(NVHandleDescArray *self, NVHandleDesc *desc);
+#define nvhandle_desc_array_index(self, i)      (((NVHandleDesc*) (self)->data) [(i)])
```

Comment:
I would expect `nvhandle_desc_array_index()` to return with `NVHandleDesc`, though on the caller site we only need a pointer to it. However with inline function one cannot write `&nvhandle_desc_array_index(...)` because one cannot take the address of an rvalue. With the `#define` version, `&nvhandle_desc_array_index(...)` will work.

Alternatively we could write an inline function that returns a pointer, but then we need to find better name for the function.  Still for dereferencing, me seems more natural to return the whole object than a pointer to it.

So in this case imo the `#define` is better.

Reply:
Ok.